client/protocol.js

```diff
@@ -0,0 +1,140 @@
+    //
+    // Game.Socket uses Transport to exchange messages with the server, and then
+    // dispatches them as events (with the ID being the same as in the spec).
+    // TODO validation maybe; could use JSON Schema also
+    //
+    // Actual handling logic is in game.js.
+    //
+
+    Game.Socket = function(transport) {
+        var self = this;
+        var messages = {};
+
+        messages.welcome = function(msg) {
+            return {
+                id: msg.id,
+                speed: msg.speed,
+                size: msg.size,
+                bulletSpeed: msg.bullet_speed,
+                bulletSize: msg.bullet_size,
+            };
+        };
+
+        messages.go_away = function(msg) {
+            transport.disconnect();
+            return {
+                reason: msg.reason,
+            };
+        };
+
+        messages.player_joined = messages.player_left = function(msg) {
+            return {
+                id: msg.id
+            };
+        };
+
+        messages.shots_fired = function(msg) {
+            return {
+                id: msg.id,
+                bulletID: msg.bullet_id,
+                position: Game.makeVector(msg.x, msg.y),
+                aim: Game.makeVector(msg.aim_x, msg.aim_y),
+            };
+        };
+
+        messages.player_spawned = function(msg) {
+            return {
+                id: msg.id,
+                position: Game.makeVector(msg.x, msg.y),
+            };
+        };
+
+        messages.player_destroyed = function(msg) {
+            return {
+                id: msg.id,
+                killerID: msg.killer_id || null,
+                bulletID: msg.bullet_id || null,
+            };
+        };
+
+        messages.player_moving = function(msg) {
+            return {
+                id: msg.id,
+                position: Game.makeVector(msg.x, msg.y),
+                direction: Game.makeVector(msg.move_x, msg.move_y),
+            };
+        };
+
+        messages.player_stopped = function(msg) {
+            return {
+                id: msg.id,
+                position: Game.makeVector(msg.x, msg.y),
+            };
+        };
+
+        messages.world_state = function(msg) {
+            var entityMap = function(entity) {
+                var result = {
+                    id: entity.id,
+                    position: Game.makeVector(entity.x, entity.y),
+                };
+
+                if (entity.move_x !== undefined && entity.move_y !== undefined) {
+                    result.direction = Game.makeVector(entity.move_x, entity.move_y);
+                }
+
+                return result;
+            };
+
+            return {
+                playerCount: msg.player_count,
+                alivePlayers: msg.alive_players.map(entityMap),
+                aliveBullets: msg.alive_bullets.map(entityMap),
+            };
+        };
+
+        var dispatch = function(msg) {
+            var handleMessage = messages[msg.type];
```

Comment:
`handleMessage` is the message handler right? If so `messageHandler` would be a more apt name.